deployment.md

```diff
@@ -0,0 +1,162 @@
+# ERC20 paymaster deployment
+
+This document describes the deployment process of the ERC20 paymaster and the accompanying infrastructure.
+
+## Prerequisites
+
+### Foundry
+
+Deployments are done using the `foundry` toolchain. Make sure you have it installed.
+
+Read more about installation [on their official documentation](https://book.getfoundry.sh/getting-started/installation).
+
+### Dependencies
+
+ERC20 paymaster uses standalone contracts for its operation:
+
+- 2 Oracles, that provide the price feed for the paymaster
+- Uniswap-v3-compatible liquidity pool
+- ERC20Paymaster itself
+
+#### Oracles
+
+Note that 2 oracles provide the Token <> USD and USD <> ETH price feeds, which require 2 corresponding liquidity pools.
+However, it is possible to operate with only 1 liquidity pool, Token <> ETH. In this case, one oracle will provide the Token <> ETH price, and the other should state the exchange rate of USD <> USD, and therefore can be a fixed oracle.
+
+#### Liqiudity pool
+
+Token <> ETH liquidity pool is required for the operation of the ERC20 paymaster. It is used by the TwapOracle to provide the price feed for the ERC20 paymaster.
+Liqiudity pool deployment is outside the scope of this document.
```

Comment:
Consider highlighting requirements for the pool: it should be a Uniswap v3 pools (or any v3-based alternative), specifically it should implement `IUniswapV3Pool` interface. As previous tests have showed us, it's impossible to use Uniswap v2 pools or any other pool implementations without modifying `TwapOracle` source code.

Reply:
60879dbb5b1bb13b48bb5bc131b6e990d3862a88